### ISSUE TITLE
libquicktime1-1.2.4-10: error: implicitly declaring library function 'memcpy'

### ISSUE THREAD

**Issue by cooljeanius** [fink/fink-distributions#1091]
Some more cases of `-Werror=implicit-function-declaration` being on by default again:
```
Making all in png
make[3]: Entering directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4/plugins/png'
/bin/sh ../../libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include  -I/opt/sw/include -I/opt/sw/include/libpng16 -DLOCALE_DIR=\"/opt/sw/share/locale\" -march=core2 -mtune=core2 -O3 -funroll-all-loops -fomit-frame-pointer  -finline-functions -Wall -Winline -Wmissing-declarations -Wdeclaration-after-statement -fvisibility=hidden -MT qtpng.lo -MD -MP -MF .deps/qtpng.Tpo -c -o qtpng.lo qtpng.c
/bin/sh ../../libtool --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include  -I/opt/sw/include -I/opt/sw/include/libpng16 -DLOCALE_DIR=\"/opt/sw/share/locale\" -march=core2 -mtune=core2 -O3 -funroll-all-loops -fomit-frame-pointer  -finline-functions -Wall -Winline -Wmissing-declarations -Wdeclaration-after-statement -fvisibility=hidden -MT lqt_png.lo -MD -MP -MF .deps/lqt_png.Tpo -c -o lqt_png.lo lqt_png.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -I/opt/sw/include -I/opt/sw/include/libpng16 -DLOCALE_DIR=\"/opt/sw/share/locale\" -march=core2 -mtune=core2 -O3 -funroll-all-loops -fomit-frame-pointer -finline-functions -Wall -Winline -Wmissing-declarations -Wdeclaration-after-statement -fvisibility=hidden -MT qtpng.lo -MD -MP -MF .deps/qtpng.Tpo -c qtpng.c  -fno-common -DPIC -o .libs/qtpng.o
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I../.. -I../../include -I../../include -I/opt/sw/include -I/opt/sw/include/libpng16 -DLOCALE_DIR=\"/opt/sw/share/locale\" -march=core2 -mtune=core2 -O3 -funroll-all-loops -fomit-frame-pointer -finline-functions -Wall -Winline -Wmissing-declarations -Wdeclaration-after-statement -fvisibility=hidden -MT lqt_png.lo -MD -MP -MF .deps/lqt_png.Tpo -c lqt_png.c  -fno-common -DPIC -o .libs/lqt_png.o
clangclang: : warningwarning: : optimization flag '-funroll-all-loops' is not supported [-Wignored-optimization-argument]optimization flag '-funroll-all-loops' is not supported [-Wignored-optimization-argument]

qtpng.c:74:5: error: implicitly declaring library function 'memcpy' with type 'void *(void *, const void *, unsigned long)' [-Werror,-Wimplicit-function-declaration]
    memcpy(data, codec->buffer + codec->buffer_position, length);
    ^
qtpng.c:74:5: note: include the header <string.h> or explicitly provide a declaration for 'memcpy'
qtpng.c:213:7: error: implicitly declaring library function 'strcasecmp' with type 'int (const char *, const char *)' [-Werror,-Wimplicit-function-declaration]
  if(!strcasecmp(key, "png_compression_level"))
      ^
qtpng.c:213:7: note: include the header <strings.h> or explicitly provide a declaration for 'strcasecmp'
2 errors generated.
make[3]: *** [Makefile:400: qtpng.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
mv -f .deps/lqt_png.Tpo .deps/lqt_png.Plo
make[3]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4/plugins/png'
make[2]: *** [Makefile:371: all-recursive] Error 1
make[2]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4/plugins'
make[1]: *** [Makefile:444: all-recursive] Error 1
make[1]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4'
make: *** [Makefile:351: all] Error 2
### execution of /tmp/fink.XEm7v failed, exit code 2
### execution of /tmp/fink.ZAdrf failed, exit code 2
Removing runtime build-lock...
Removing build-lock package...
/opt/sw/bin/dpkg-lockwait -r fink-buildlock-libquicktime1-1.2.4-10
(Reading database ... 
dpkg: serious warning: files list file for package `fink-buildlock-ccp4-6.5-1' missing, assuming package has no files currently installed.
1029151 files and directories currently installed.)
Removing fink-buildlock-libquicktime1-1.2.4-10 ...
```
(I'm on x86_64 Big Sur with Xcode 13)

**Comment by cooljeanius** [comment]
OK so af67fde fixed the implicit function declaration errors I originally reported here; now I'm getting a bunch of missing ogg symbols, though:
```
/bin/sh ../../libtool --tag=CC   --mode=link gcc -I/opt/sw/include -DLOCALE_DIR=\"/opt/sw/share/locale\" -march=core2 -mtune=core2 -O3 -funroll-all-loops -fomit-frame-pointer  -finline-functions -Wall -Winline -Wmissing-declarations -Wdeclaration-after-statement -fvisibility=hidden -avoid-version -module -L/opt/sw/lib -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names -o lqt_vorbis.la -rpath /opt/sw/lib/libquicktime1 vorbis.lo lqt_vorbis.lo ../../src/libquicktime.la -lvorbisenc -L/opt/sw/lib -lvorbis -lm -lvorbisfile -lm -lz -ldl 
libtool: link: gcc  -o .libs/lqt_vorbis.so -bundle  .libs/vorbis.o .libs/lqt_vorbis.o   -L/opt/sw/lib ../../src/.libs/libquicktime.dylib /opt/sw/lib/libvorbisenc.dylib /opt/sw/lib/libvorbis.dylib /opt/sw/lib/libvorbisfile.dylib -lm -lz -ldl  -march=core2 -mtune=core2 -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names  
Undefined symbols for architecture x86_64:
  "_ogg_page_eos", referenced from:
      _flush_data in vorbis.o
  "_ogg_page_serialno", referenced from:
      _next_page in vorbis.o
  "_ogg_stream_clear", referenced from:
      _delete_codec in vorbis.o
      _decode in vorbis.o
  "_ogg_stream_flush", referenced from:
      _encode in vorbis.o
      _flush_data in vorbis.o
  "_ogg_stream_init", referenced from:
      _encode in vorbis.o
      _next_page in vorbis.o
  "_ogg_stream_packetin", referenced from:
      _encode in vorbis.o
      _flush_data in vorbis.o
  "_ogg_stream_packetout", referenced from:
      _decode in vorbis.o
      _decode_frame in vorbis.o
  "_ogg_stream_pagein", referenced from:
      _next_page in vorbis.o
  "_ogg_sync_buffer", referenced from:
      _next_page in vorbis.o
  "_ogg_sync_init", referenced from:
      _decode in vorbis.o
  "_ogg_sync_pageout", referenced from:
      _next_page in vorbis.o
  "_ogg_sync_reset", referenced from:
      _decode in vorbis.o
  "_ogg_sync_wrote", referenced from:
      _next_page in vorbis.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[3]: *** [Makefile:374: lqt_vorbis.la] Error 1
make[3]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4/plugins/vorbis'
make[2]: *** [Makefile:371: all-recursive] Error 1
make[2]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4/plugins'
make[1]: *** [Makefile:444: all-recursive] Error 1
make[1]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4'
make: *** [Makefile:351: all] Error 2
### execution of /tmp/fink.xTDuT failed, exit code 2
### execution of /tmp/fink.F6Kbc failed, exit code 2
Removing runtime build-lock...
Removing build-lock package...
/opt/sw/bin/dpkg-lockwait -r fink-buildlock-libquicktime1-1.2.4-10
(Reading database ... 
dpkg: serious warning: files list file for package `fink-buildlock-ccp4-6.5-1' missing, assuming package has no files currently installed.
1029151 files and directories currently installed.)
Removing fink-buildlock-libquicktime1-1.2.4-10 ...
```

**Comment by cooljeanius** [comment]
> OK so [af67fde](https://github.com/fink/fink-distributions/commit/af67fded330612baddb0bc09d20aabc96b9334ef) fixed the implicit function declaration errors I originally reported here; now I'm getting a bunch of missing ogg symbols, though:
> 
> ```
> /bin/sh ../../libtool --tag=CC   --mode=link gcc -I/opt/sw/include -DLOCALE_DIR=\"/opt/sw/share/locale\" -march=core2 -mtune=core2 -O3 -funroll-all-loops -fomit-frame-pointer  -finline-functions -Wall -Winline -Wmissing-declarations -Wdeclaration-after-statement -fvisibility=hidden -avoid-version -module -L/opt/sw/lib -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names -o lqt_vorbis.la -rpath /opt/sw/lib/libquicktime1 vorbis.lo lqt_vorbis.lo ../../src/libquicktime.la -lvorbisenc -L/opt/sw/lib -lvorbis -lm -lvorbisfile -lm -lz -ldl 
> libtool: link: gcc  -o .libs/lqt_vorbis.so -bundle  .libs/vorbis.o .libs/lqt_vorbis.o   -L/opt/sw/lib ../../src/.libs/libquicktime.dylib /opt/sw/lib/libvorbisenc.dylib /opt/sw/lib/libvorbis.dylib /opt/sw/lib/libvorbisfile.dylib -lm -lz -ldl  -march=core2 -mtune=core2 -Wl,-headerpad_max_install_names -Wl,-headerpad_max_install_names  
> Undefined symbols for architecture x86_64:
>   "_ogg_page_eos", referenced from:
>       _flush_data in vorbis.o
>   "_ogg_page_serialno", referenced from:
>       _next_page in vorbis.o
>   "_ogg_stream_clear", referenced from:
>       _delete_codec in vorbis.o
>       _decode in vorbis.o
>   "_ogg_stream_flush", referenced from:
>       _encode in vorbis.o
>       _flush_data in vorbis.o
>   "_ogg_stream_init", referenced from:
>       _encode in vorbis.o
>       _next_page in vorbis.o
>   "_ogg_stream_packetin", referenced from:
>       _encode in vorbis.o
>       _flush_data in vorbis.o
>   "_ogg_stream_packetout", referenced from:
>       _decode in vorbis.o
>       _decode_frame in vorbis.o
>   "_ogg_stream_pagein", referenced from:
>       _next_page in vorbis.o
>   "_ogg_sync_buffer", referenced from:
>       _next_page in vorbis.o
>   "_ogg_sync_init", referenced from:
>       _decode in vorbis.o
>   "_ogg_sync_pageout", referenced from:
>       _next_page in vorbis.o
>   "_ogg_sync_reset", referenced from:
>       _decode in vorbis.o
>   "_ogg_sync_wrote", referenced from:
>       _next_page in vorbis.o
> ld: symbol(s) not found for architecture x86_64
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> make[3]: *** [Makefile:374: lqt_vorbis.la] Error 1
> make[3]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4/plugins/vorbis'
> make[2]: *** [Makefile:371: all-recursive] Error 1
> make[2]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4/plugins'
> make[1]: *** [Makefile:444: all-recursive] Error 1
> make[1]: Leaving directory '/opt/sw/src/fink.build/libquicktime1-1.2.4-10/libquicktime-1.2.4'
> make: *** [Makefile:351: all] Error 2
> ### execution of /tmp/fink.xTDuT failed, exit code 2
> ### execution of /tmp/fink.F6Kbc failed, exit code 2
> Removing runtime build-lock...
> Removing build-lock package...
> /opt/sw/bin/dpkg-lockwait -r fink-buildlock-libquicktime1-1.2.4-10
> (Reading database ... 
> dpkg: serious warning: files list file for package `fink-buildlock-ccp4-6.5-1' missing, assuming package has no files currently installed.
> 1029151 files and directories currently installed.)
> Removing fink-buildlock-libquicktime1-1.2.4-10 ...
> ```

...ok, I can confirm that @nieder successfully fixed this second part in 83a592d. (Thanks!)
